src/lib/sessions.ts

```diff
@@ -16,7 +16,7 @@ export interface Session {
 	id: string;
 	model: string;
 	messages: Message[];
-	systemPrompt: Message;
+	systemPrompt?: Message | Message[];
```

Comment:
Yeah, I did this so we can attach multiple knowledge (or images/urls in the future), but I know that it might not be the best solution. Do you have a better idea?

Reply:
Oh sorry, I missed this in my review.
You are submitting the "Knowledge" as a system prompt, I think we should submit it as a `role: 'user'` instead:

```json
[
  {
    "role": "user",
    "knowledge": {
      "id": "82bcfb",
      "name": "Susan",
      "content": "You are a helpful AI assistant made by Hollama.\nYour name is Susan.\nAnwer the user prompts the best you can.",
      "updatedAt": "2024-11-07T22:13:28.779Z"
    },
    "content": "You are a helpful AI assistant made by Hollama.\nYour name is Susan.\nAnwer the user prompts the best you can."
  },
  {
    "role": "user",
    "content": "hi, i'm steve"
  }
]
```

I think we can send multiple `user` messages at once but I don't think we can send multiple `system` prompts.

Reply:
Great then! I'll apply the necessary changes.

Reply:
Now that I think about it, do we need the `systemPrompt` at all?

Reply:
We do, the `systemPrompt` it's a special type of message:

> System prompts are a special type of instruction used to set up the behavior of the model. It defines how the model should behave throughout the conversation. This prompt is usually provided at the start of the session to guide the tone, personality, and overall style of the AI’s responses.